tests/test_transpile.py

```diff
@@ -1004,6 +1004,60 @@ def test_transpile_expressions(here, builder, frontend, use_c_ptr):
     f2c.wrapperpath.unlink()
     f2c.c_path.unlink()
 
+
+@pytest.mark.parametrize('use_c_ptr', (False, True))
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_transpile_call(here, frontend, use_c_ptr):
+    fcode_module = """
+module transpile_call_kernel_mod
+  implicit none
+contains
+
+  subroutine transpile_call_kernel(a, b, c, arr1, len)
+    integer, intent(inout) :: a, c
+    integer, intent(in) :: b
+    integer, intent(in) :: len
+    integer, intent(inout) :: arr1(len, len)
```

Comment:
For my education: What is the intended outcome if we had (another) argument `arr(:,:)` here? Would this pre-empt the F2C transformation until we figure out the actual shape, or would it simply not be passed as a reference?

Reply:
Since we need the actual `shape` information for flattening (see `flatten_arrays()` - *TypeError: Resolve shapes being of type RangeIndex, e.g., ":" before flattening!*, raised if `isinstance(shape[-2], sym.RangeIndex)`)arrays having more than one dimension, the `DeReferenceTrafo` wouldn't be executed.

However, 

```fortran
subroutine transpile_call_driver(a)
  use transpile_call_kernel_mod, only: transpile_call_kernel
    integer, parameter :: len = 5
    integer, intent(inout) :: arr1(len)
    integer, intent(inout) :: arr2(len)
    call transpile_call_kernel(arr1, arr2, len)
end subroutine transpile_call_driver

  subroutine transpile_call_kernel(arr1, arr2, len)
    integer, intent(in) :: len
    integer, intent(inout) :: arr1(len)
    integer, intent(inout) :: arr2(:)

    arr1(1) = 1
    arr2(1) = 1
  end subroutine transpile_call_kernel
```

is transformed/transpiled to:

```c
int transpile_call_driver_c() {
  int len = 5;
  transpile_call_kernel(arr1, arr2, len);
  return 0;
}

int transpile_call_kernel_c(int * restrict arr1, int * restrict arr2, int len) {

  int arr1[len];
  int arr2[len];
  arr1[1 - 1] = 1;
  arr2[1 - 1] = 1;
  return 0;
}
```

if that answers your question?!

Reply:
It does, many thanks!